Gek62/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+def data
+  @data = File.read('data', encoding: 'UTF-8')
+end
+
+def blank_error
+  puts 'City name can not be blank!'
+end
+
+def mean(town, data)
+  return -1 if (numbers = extract_data(town, data)).nil?
+
+  numbers.sum / 12
+end
+
+def variance(town, data)
+  return -1 if (numbers = extract_data(town, data)).nil?
+
+  mean = numbers.sum / 12
+  @variance = numbers.map { |x| (x - mean)**2 }.sum / 12
+end
+
+def extract_data(town, data)
+  data_hash = data.split("\n").map { |str| str.split(':') }.to_h
```

Comment:
As a more concise option:
```suggestion
  data_hash = data.lines.map { |str| str.split(':') }.to_h
```